pkg/webui/console/store/middleware/logics/lib.js

```diff
@@ -79,41 +110,47 @@ const createRequestLogic = function(
         if (promiseAttached) {
           const {
             meta: { _resolve },
-          } = deps.action
+          } = action
           _resolve(res)
         }
       } catch (e) {
         error(e) // Log the error if in development mode.
-
         if (isUnauthenticatedError(e)) {
           // If there was an unauthenticated error, the access token is not
           // valid and we can delete it. Reloading will then initiate the auth
           // flow.
           clearAccessToken()
           window.location.reload()
-        } else {
-          // Otherwise, dispatch the fail action and report it to Sentry.
-          if (isInvalidArgumentError(e)) {
-            Sentry.withScope(scope => {
-              scope.setExtras(e)
-              const fingerprint = getBackendErrorId(e)
-              scope.setFingerprint(fingerprint)
-              Sentry.captureException(new Error(fingerprint))
-            })
-          } else if (isUnknown(e)) {
-            Sentry.withScope(scope => {
-              scope.setExtras(e)
-              Sentry.captureException(e)
-            })
+        } else if (isNetworkError(e) || isTimeoutError(e)) {
+          // If there was a network error, it could mean that the network
+          // connection is currently interrupted. Setting the online state to
+          // `checking` will trigger respective connection management logics.
+          if (!selectIsCheckingStatus(getState()) && action.type !== ATTEMPT_RECONNECT) {
+            // We only need to set the status and trigger the connection checks
+            // if the online status was `online` previously.
+            dispatch(setOnlineStatus(ONLINE_STATUS.CHECKING))
```

Comment:
```suggestion
            dispatch(setStatusChecking())
```

Not sure if `setOnlineStatus` is even needed